paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/PaimonRecordReaderIterator.scala

```diff
@@ -84,7 +95,18 @@ case class PaimonRecordReaderIterator(reader: RecordReader[PaimonInternalRow])
       try {
         var stop = false
         while (!stop) {
-          currentResult = currentIterator.next
+          val dataRow = currentIterator.next()
+          if (dataRow != null) {
+            if (needMetadata && isFileRecordIterator) {
+              updateMetadataRow(currentIterator.asInstanceOf[FileRecordIterator[PaimonInternalRow]])
+              currentResult = joinedRow.replace(dataRow, metadataRow)
+            } else {
+              currentResult = dataRow
```

Comment:
needMetadata & is not FileRecordIterator
Here should throw exception?

Reply:
I prefer to return null instead of throwing exception.